pinot-common/src/main/java/org/apache/pinot/pql/parsers/Pql2Compiler.java

```diff
@@ -103,6 +110,24 @@ public BrokerRequest compileToBrokerRequest(String expression)
 
       BrokerRequest brokerRequest = new BrokerRequest();
       rootNode.updateBrokerRequest(brokerRequest);
+      try {
+        PinotQuery pinotQuery = new PinotQuery();
+        rootNode.updatePinotQuery(pinotQuery);
+        boolean validateConverter = true;
+        if (validateConverter) {
```

Comment:
This is the switch described in the doc for us to ensure we could reconstruct a PinotBroker from a parsed PinotQuery.
I will make it a system env, so that we could turn it on/off in staging/test env.